alacritty_terminal/src/term/mod.rs

```diff
@@ -1153,15 +1154,50 @@ impl<T> Term<T> {
             num_lines = Line(2);
         }
 
+        // Scroll up to keep cursor in terminal
+        if self.cursor.point.line >= num_lines {
+            let lines = self.cursor.point.line - num_lines + 1;
+            let template = Cell { bg: self.cursor.template.bg, ..Cell::default() };
+            self.grid.scroll_up(&(Line(0)..old_lines), lines, &template);
+        }
+
+        // Scroll up inactive grid as well
+        if self.cursor_save_inactive.point.line >= num_lines {
+            let lines = self.cursor_save_inactive.point.line - num_lines + 1;
+            let template = Cell { bg: self.cursor_save_inactive.template.bg, ..Cell::default() };
+            self.inactive_grid.scroll_up(&(Line(0)..old_lines), lines, &template);
+        }
+
+        // Move prompt down when growing if scrollback lines are available
+        if num_lines > old_lines && !self.mode.contains(TermMode::ALT_SCREEN) {
+            let growage = min(num_lines - old_lines, Line(self.grid.history_size()));
+            self.cursor.point.line += growage;
+        }
+
```

Comment:
Looks like you didn't rebase this correctly. This code does not exist on the latest master.

Reply:
Something did go horribly wrong it seems. I'll clean this up.